sunpy/io/fits.py

```diff
@@ -200,19 +200,25 @@ def header_to_fits(header):
     key_comments = header.pop('KEYCOMMENTS', False)
 
     for k, v in header.items():
-        if isinstance(v, fits.header._HeaderCommentaryCards):
-            if k.upper() == 'COMMENT':
-                comments = str(v).split('\n')
-                for com in comments:
-                    fits_header.add_comment(com)
-            elif k.upper() == 'HISTORY':
-                hists = str(v).split('\n')
-                for hist in hists:
-                    fits_header.add_history(hist)
-            elif k != '':
+        if len(k) > 8:
```

Comment:
Add a comment explaining the (currently) seemingly arbitrary 8?

Reply:
We are saving to FITS headers, you can't have keys in a FITS header longer than 8.